opm/autodiff/NonlinearSolver.hpp

```diff
@@ -169,10 +169,13 @@ namespace Opm {
         double relaxRelTol() const       { return param_.relax_rel_tol_; }
 
         /// The maximum number of nonlinear iterations allowed.
-        int maxIter() const           { return param_.max_iter_; }
+        int maxIter() const              { return param_.max_iter_; }
 
         /// The minimum number of nonlinear iterations allowed.
-        double minIter() const           { return param_.min_iter_; }
+        int minIter() const              { return param_.min_iter_; }
+
+        /// Set parameters to override those given at construction time.
+        void setParameters(const SolverParameters& param) { param_ = param; }
```

Comment:
They are not, so I think the move-less approach is ok. Also the objects are small and few (2 in a run)/